### PR TITLE
Don't fetch parts from lost replicas

### PR DESCRIPTION
**Pull request by alesapin**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add Merge tree setting `fetch_parts_from_lost_replicas` which restrict replicas to search and fetch parts from other replicas which are marked as lost.

### CODE REVIEW

**Comment by tavplubix** [comment]
It's questionable, because after #30346 we take into account future parts of source replica when deciding which parts should be removed to recover replica. So it makes sense to wait for lost replica to become active again. 

But maybe it's ok if the setting is disabled by default.

File: src/Storages/MergeTree/MergeTreeSettings.h [hunk 1]
```diff
@@ -130,6 +130,7 @@ struct Settings;
     M(UInt64, min_marks_to_honor_max_concurrent_queries, 0, "Minimal number of marks to honor the MergeTree-level's max_concurrent_queries (0 - disabled). Queries will still be limited by other max_concurrent_queries settings.", 0) \
     M(UInt64, min_bytes_to_rebalance_partition_over_jbod, 0, "Minimal amount of bytes to enable part rebalance over JBOD array (0 - disabled).", 0) \
     M(Bool, check_sample_column_is_correct, true, "Check columns or columns by hash for sampling are unsigned integer.", 0) \
+    M(Bool, fetch_parts_from_lost_replicas, true, "Try to download parts from replicas marked as lost (is_lost flag in zookeeper)", 0) \
```
Comment (tavplubix) [inline src/Storages/MergeTree/MergeTreeSettings.h:133]:
```suggestion
    M(Bool, fetch_parts_from_lost_replicas, true, "Try to download parts from replicas marked as lost (is_lost flag in zookeeper). Disabling this setting may cause data loss in rare cases", 0) \
```

Comment (alesapin) [reply src/Storages/MergeTree/MergeTreeSettings.h:133]: :D

**Comment by alesapin** [comment]
Better not to have it.

**Pull request closed without merging**